java/core/src/java/org/apache/orc/impl/ColumnStatisticsImpl.java

```diff
@@ -170,7 +165,198 @@ public int hashCode() {
+        minimum = length;
+      } else if (length > maximum) {
+        maximum = length;
+      }
+
+      this.sum += length;
+    }
+
+    @Override
+    public void reset() {
+      super.reset();
+      hasMinimum = false;
+      minimum = Long.MAX_VALUE;
+      maximum = Long.MIN_VALUE;
+      sum = 0;
+      overflow = false;
+    }
+
+    @Override
+    public void merge(ColumnStatisticsImpl other) {
+      if (other instanceof CollectionColumnStatisticsImpl) {
+        CollectionColumnStatisticsImpl otherColl = (CollectionColumnStatisticsImpl) other;
+        if (!hasMinimum) {
+          hasMinimum = otherColl.hasMinimum;
+          minimum = otherColl.minimum;
+          maximum = otherColl.maximum;
+        } else if (otherColl.hasMinimum) {
+          if (otherColl.minimum < minimum) {
+            minimum = otherColl.minimum;
+          }
+          if (otherColl.maximum > maximum) {
+            maximum = otherColl.maximum;
+          }
+        }
+
+        overflow |= otherColl.overflow;
+        if (!overflow) {
+          boolean wasPositive = sum >= 0;
+          sum += otherColl.sum;
+          if ((otherColl.sum >= 0) == wasPositive) {
+            overflow = (sum >= 0) != wasPositive;
+          }
+        }
+      } else {
+        if (isStatsExists() && hasMinimum) {
+          throw new IllegalArgumentException("Incompatible merging of collection column statistics");
+        }
+      }
+      super.merge(other);
+    }
+
+    @Override
+    public long getMinimumChildren() {
+      return minimum;
+    }
+
+    @Override
+    public long getMaximumChildren() {
+      return maximum;
+    }
+
+    @Override
+    public long getTotalChildren() {
+      return sum;
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder buf = new StringBuilder(super.toString());
+      if (hasMinimum) {
+        buf.append(" minChildren: ");
+        buf.append(minimum);
+        buf.append(" maxChildren: ");
+        buf.append(maximum);
+      }
+      if (!overflow) {
+        buf.append(" totalChildren: ");
+        buf.append(sum);
+      }
+      return buf.toString();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (!(o instanceof CollectionColumnStatisticsImpl)) {
+        return false;
+      }
+      if (!super.equals(o)) {
+        return false;
+      }
+
+      CollectionColumnStatisticsImpl that = (CollectionColumnStatisticsImpl) o;
+
+      if (minimum != that.minimum) {
+        return false;
+      }
+      if (maximum != that.maximum) {
+        return false;
+      }
+      if (sum != that.sum) {
+        return false;
+      }
+      if (hasMinimum != that.hasMinimum) {
+        return false;
+      }
+      if (overflow != that.overflow) {
+        return false;
+      }
+
+      return true;
+    }
+
+    @Override
+    public int hashCode() {
+      int result = super.hashCode();
```

Comment:
I'd suggest using a simpler hashCode if the count is zero rather than hashing in the Long MIN and MAX.

Reply:
Sure, fixed it.